### PR TITLE
Enable forall loops over zippered loop expressions

### LINKED ISSUE #10282: enable forall loop over forall expr over zip(A,B)
The following code currently crashes the compiler,
while we agree that it should work.

```chpl
var A, B: [1..4] int;
forall xy in ([ab in zip(A,B)] ab) do
  xy = (1,5); // expected to update A and B's elements
```

The cause appears to be inconsistency of what the "zippered" flag means in the ForallStmt node.

test:
```
test/expressions/loop-expr/forall-over-zip-over-arrays.chpl
```

### PR DESCRIPTION
**Pull request by vasslitvinov**
If we have a loop expression in the leader position of a forall statement,
in the AST it is a call to chpl__loopexpr_iter(). It is converted by
convertIteratorForLoopexpr() to call directly the leader iterator these().

When the loop expression itself is zippered, chpl__loopexpr_iter() accepts
a tuple of all things within the zip() syntax. However, the leader iterator
these() accepts only the first one of those things.

This PR adds an adjustment to handle this situation, by extracting this first
thing from the tuple.

Long term, we need to design the AST to eliminate the need in
convertIteratorForLoopexpr() altogether.

De-futurize and beef up the corresponding test.
Let it be tested in all configurations.

Testing: linux64.

Resolves #10282.

### CODE REVIEW

**Comment by vasslitvinov** [comment]
@benharsh or @mppf would you review?

File: compiler/AST/foralls.cpp [hunk 1]
```diff
@@ -1116,8 +1116,27 @@ static void removeOuterVarArgs(CallExpr* iterCall, FnSymbol* oldCallee,
   }
 }
 
+// If the first iterable is a zippered loop expr, convertIteratorForLoopexpr()
+// will plug in the leader. Which accepts just the first comp of the tuple.
```
Comment (mppf) [inline compiler/AST/foralls.cpp]: comp -> component

File: compiler/AST/foralls.cpp [hunk 2]
```diff
@@ -1116,8 +1116,27 @@ static void removeOuterVarArgs(CallExpr* iterCall, FnSymbol* oldCallee,
...
+// Adjust the call correspondingly.
```
Comment (mppf) [inline compiler/AST/foralls.cpp]: Let's put a comment here describing the pattern we're trying to support.

File: compiler/AST/foralls.cpp [hunk 3]
```diff
@@ -1116,8 +1116,27 @@ static void removeOuterVarArgs(CallExpr* iterCall, FnSymbol* oldCallee,
...
+  Symbol* firstField = ag->getField(1);
+  VarSymbol* temp = newTemp("firstField", firstField->type);
+  firstArg->replace(new SymExpr(temp));
+  stmt->insertBefore(new DefExpr(temp));
+  stmt->insertBefore("'move'(%S,'.v'(%E,%S))", temp, firstArg, firstField);
+}
+
 //
-// Handle the case where the leader iterator is astr_loopexpr_iter.
+// Handle the case where the leader iterator is _iterator_for_loopexpr.
```
Comment (mppf) [inline compiler/AST/foralls.cpp]: I'm a little suprised the method below doesn't have LeaderIterator in the name somewhere, but I havn't dug in to the call site.

**Pull request merged**